### PR TITLE
BL-2310 Make equal left/right margins for calendar fold

### PR DESCRIPTION
**Pull request by gmartin7**
* only applies to: a)  A4Landscape (on A3 paper)
   and b) A5Landscape (on A4 paper)
* the latter only being found (so far) in Wall Calendar,
   the calendar bottom page needed an extra tweak.

### CODE REVIEW

File: src/BloomBrowserUI/bookLayout/basePage.less [hunk 1]
```diff
@@ -257,17 +257,27 @@ preview.css.
 //currently, in edit mode every page looks like page 1 to css, so left-vs-right margins are just confusing
 //so we only turn them on when making PDF
 .publishMode :not(.outsideFrontCover):not(.outsideBackCover){
-  &.bloom-page:nth-of-type(odd) .marginBox  {
-	  left: @MarginOuter;
-  }
-  &.bloom-page:nth-of-type(even) .marginBox  {
-	  left: @MarginInner;
-  }
+	&.bloom-page:nth-of-type(odd){
```
Comment (hatton) [inline src/BloomBrowserUI/bookLayout/basePage.less]: These new rules specify particular page size/layouts, where the old doesn't. So for example an A6 book would have been served by the old rule, but now would have no left margin.

File: src/BloomBrowserUI/bookLayout/basePage.less [hunk 2]
```diff
@@ -257,17 +257,27 @@ preview.css.
...
+		&.A5Landscape:not(.calendarMonthBottom) .marginBox, &.A4Landscape .marginBox{
+			left: @MarginEven; // BL-2310 keep left and right the same for calendar fold
+		}
+	}
 }
 
 //in edit mode, just split the difference, centering the margin box
 body:not(.publishMode) .marginBox{
-  left: @MarginOuter + ( (@MarginInner - @MarginOuter) / 2)
+  left: @MarginEven;
```
Comment (hatton) [inline src/BloomBrowserUI/bookLayout/basePage.less]: Is this related to the purpose of the PR? The old code was clearly correct, the new? no idea, except that it doesn't fit the comment.

File: DistFiles/less/common-mixins.less [hunk 3]
```diff
@@ -38,4 +38,5 @@
 @MarginOuter: 15mm;
 @MarginBottom: 15mm;
 @ExtraSpaceForBinding: 10mm;
-@MarginInner: @MarginOuter + @ExtraSpaceForBinding;
\ No newline at end of file
+@MarginInner: @MarginOuter + @ExtraSpaceForBinding;
+@MarginEven: (@MarginOuter + @MarginInner) / 2.0; // for when we want the same left and right margins
```
Comment (hatton) [inline DistFiles/less/common-mixins.less]: When I read this below, I read "even" to mean "opposite of odd". How about "MarginBothSidesSame"

Comment (JohnThomson) [inline DistFiles/less/common-mixins.less]: Or MarginBalanced or MarginEqual if you want something shorter

File: src/BloomBrowserUI/bookLayout/basePage.less (hunk 1 above)
Comment (gmartin7) [inline src/BloomBrowserUI/bookLayout/basePage.less]: The A6 booklets seem to be folded "horizontally" (i.e. not calendar-folded) for both Portrait and Landscape, so they fall under the :not(.A5Landscape):not(.A4Landscape) sections and get Outer or Inner margins like the other non-Landscape layouts.

File: DistFiles/less/common-mixins.less (hunk 3 above)
Comment (gmartin7) [inline DistFiles/less/common-mixins.less]: Good point, "Johns". I'll go with MarginBalanced.

File: src/BloomBrowserUI/bookLayout/basePage.less (hunk 1 above)
Comment (hatton) [inline src/BloomBrowserUI/bookLayout/basePage.less]: But of course :-) Alternatively, I often don't try to exclude the special cases, as that makes it hard to read. Instead, you can make use of the 2 rules that would make your special case win: it is more specific, and it comes later.

So you could just have 

    .marginBox { left: @MarginOuter; }

followed by your new rule:

    // override that for the special cases of A5Landscape and A4Landscape. A6Landscape is not included here because....

Comment (gmartin7) [inline src/BloomBrowserUI/bookLayout/basePage.less]: OK. I was able to simplify these rules as you mention, but I needed to make one rule in wallCalendar "important", since it applies in both Edit and Publish mode and is specific to the wall calendar.
